src/secrets-kubernetes/src/lib.rs

```diff
@@ -100,12 +137,40 @@ impl SecretsController for KubernetesSecretsController {
 
         self.secret_api
             .patch(
-                &SECRET_NAME,
+                &self.config.user_defined_secret,
                 &PatchParams::apply(FIELD_MANAGER).force(),
                 &Patch::Apply(secret),
             )
             .await?;
 
+        self.trigger_resync().await?;
+
+        // guarantee that all new secrets are reflected on our local filesystem
```

Comment:
I have to verify all of this manually, so don't consider this code final until I've resolved this comment.